src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+import uk.gov.companieshouse.web.accounts.service.smallfull.RptTransactionService;
+import uk.gov.companieshouse.web.accounts.transformer.smallfull.relatedpartytransactions.RptTransactionsTransformer;
+import uk.gov.companieshouse.web.accounts.util.ValidationContext;
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+import uk.gov.companieshouse.web.accounts.validation.helper.ServiceExceptionHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class RptTransactionsServiceImpl implements RptTransactionService {
+
+    @Autowired
+    private ApiClientService apiClientService;
+
+    @Autowired
+    private RptTransactionsTransformer rptTransactionsTransformer;
+
+    @Autowired
+    private ServiceExceptionHandler serviceExceptionHandler;
+
+    @Autowired
+    private ValidationContext validationContext;
+
+    private static final UriTemplate RPT_TRANSACTIONS_URI =
+            new UriTemplate("/transactions/{transactionId}/company-accounts/{companyAccountsId}/small-full/notes/related-party-transactions/transactions");
+
+    private static final UriTemplate RPT_TRANSACTIONS_URI_WITH_ID =
+            new UriTemplate("/transactions/{transactionId}/company-accounts/{companyAccountsId}/small-full/notes/related-party-transactions/transactions/{transactionId}");
+
+    private static final String RESOURCE_NAME = "transactions";
+
+    @Override
+    public RptTransaction[] getAllRptTransactions(String transactionId, String companyAccountsId) throws ServiceException {
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        String uri = RPT_TRANSACTIONS_URI.expand(transactionId, companyAccountsId).toString();
+
+        try {
+            RptTransactionApi[] rptTransactions = apiClient.smallFull().relatedPartyTransactions().rptTransactions().getAll(uri).execute().getData();
+            return rptTransactionsTransformer.getAllRptTransactions(rptTransactions);
+        } catch (ApiErrorResponseException e) {
+            serviceExceptionHandler.handleRetrievalException(e, RESOURCE_NAME);
+        } catch (URIValidationException e) {
+            serviceExceptionHandler.handleURIValidationException(e, RESOURCE_NAME);
+        }
+
+        return new RptTransaction[0];
+    }
+
+    @Override
+    public List<ValidationError> createRptTransaction(String transactionId, String companyAccountsId, RptTransactionToAdd rptTransactionToAdd) throws ServiceException {
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        String uri = RPT_TRANSACTIONS_URI.expand(transactionId, companyAccountsId).toString();
+
+        RptTransactionApi rptTransactionApi = rptTransactionsTransformer.getRptTransactionsApi(rptTransactionToAdd);
+
+        try {
+            ApiResponse<RptTransactionApi> apiResponse = apiClient.smallFull().relatedPartyTransactions().rptTransactions().create(uri, rptTransactionApi).execute();
```

Comment:
Sonar is complaining about this apiResponse not being used. Should it be returned or used?

Reply:
Removed unused variables